python/whylogs/api/logger/result_set.py

```diff
@@ -475,6 +478,27 @@ def get_writables(self) -> Optional[List[Writable]]:
             )
         return results
 
+    def in_memory_zip(self) -> Optional[bytes]:
```

Comment:
True, in case we're dealing with a gigantic SegmentedResultSet, which can contain multiple SegmentedDatasetProfiles with a number of columns.
I'll turn it into a static method on WhyLabsWriter then, so other types of Writables can leverage it

Reply:
it will be an in-memory zipped file, that will be persisted to tmp_dir. same logic as before, only extracted out to its own method